### PR TITLE
over: work with maps and records

### PR DESCRIPTION
**Pull request by mattnibs**
Make over work with maps and records. For each element in an record/map
over will emit `{key:<key>,value:<value>}` as the context.

Also:
- Use un-union'd values for over.

### CODE REVIEW

File: proc/traverse/over.go [hunk 1]
```diff
@@ -65,26 +68,52 @@ func (o *Over) over(batch zbuf.Batch, this *zed.Value) zbuf.Batch {
...
+			// this copy through the batch.
+			val := expr.ValueUnder(zed.NewValue(typ, it.Next()))
+			vals = append(vals, *val.Copy())
+		}
 		return vals
+	case *zed.TypeMap:
+		rtyp := zctx.MustLookupTypeRecord([]zed.Column{
+			zed.NewColumn("key", typ.KeyType),
+			zed.NewColumn("value", typ.ValType),
+		})
```
Comment (mattnibs) [inline proc/traverse/over.go:93]: Hmm, I'm thinking value should also be de-unioned here as we do with arrays?

Comment (mattnibs) [reply proc/traverse/over.go:93]: Actually also the key as well.

**Comment by mccanne** [comment]
Did you want to merge this and fix the union problem later?  Might be easier after we update other parts of the system to handle unions more uniformly...

**Comment by mattnibs** [comment]
> Did you want to merge this and fix the union problem later? Might be easier after we update other parts of the system to handle unions more uniformly...

Sure!

**Review by mccanne — APPROVED** [review]

**Pull request merged**